docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,209 @@
+    #FIXME: add an annotation to specify which volume this pvc should be based on
+parameters: ...
+
+----
+ 
+❶ The unique name of the template. +
+
+❷ OS identifier of the template. +
+
+❸ Indicator if this is a kubevirt template. +
+
+❹ The OfflineVirtualMachine Object. This object must appear only once in the template. +
+
+➎ OfflineVirtualMachine’s name. +
+
+➏ OfflineVirtualMachine’s resources, currently includes under requirements only memory. +
+   Memory is the minimal amount the Virtual Machine requires. +
+   These attributes are represented as parameters that will either be filled out by the user Or receive default values. +
+   
+❼ Number of cpu cores for the Virtual Machine. +
+  This attribute is represented as parameter that will either be filled out by the user Or receive a default value. +
+  
+❽ Disk contains a "volumeName" attribute, which links it to a volume +
+
+➒ Specification of a disk's volume from the disks section. It contains persistent volume claims.  
+(more about link:https://kubernetes.io/docs/concepts/storage/persistent-volumes/[Persistent Volume Claims] ) +
+        
+➓ The PersistenVolumeClaim Object that accompanies the OfflineVirtualMachine Object. +
+
+⓫ PersistenVolumeClaim’s name. +
+
+
+
+=== Labels
+
+Labels will be presented on every OCP VirtualMachineTemplate in order for ManageIQ to recognize the KubeVirt template characteristics. + 
+The label "miq.github.io/kubevirt-os" is type of OS the template refers to. +
+The label "miq.github.io/kubevirt-is-vm-template" will indicate with a boolean value if it is a VirtualMachineTemplate.
+
+[source,yaml]
+----
+kind: Template
+metadata:
+  name: RHEL-7
+  labels:
+    miq.github.io/kubevirt-os: rhel-7 
+    miq.github.io/kubevirt-is-vm-template: true
+----
+
+=== Parameters
+
+Parameters allow a value to be supplied by the user or generated when the template is instantiated. +
+Then, that value is substituted wherever the parameter is referenced. +
+The following parameters will be used for the OfflineVirtualMachine name (and as a part of the persistent volume claim names) +
+Default values will be provided for memory and cpu, which will be used if the user does not supply a different value.
+
+[source,yaml]
+----
+parameters:
+- description: Name for the new VM     
+  name: NAME   
+- name: MEMORY
+  description: Amount of memory in MB
+  value: 512Mi  
+- name: CPU
+  description: Amount of cores
+  value: 4    
+----
+
+Additional info on link:https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/[compute resources]
+
+
+== Creating the Template object in the cluster
+
+If you have a YAML file that defines a VirtualMachine template, you can upload the template to an Openshift cluster using the CLI. 
+To upload a template to your current project’s template library, pass the YAML file with the following command: +
+
+----
+$ oc create -f <filename>
+----
+
+
+=== Creating VirtualMachineTemplate Objects using the Web Console
```

Comment:
VirtualMachineTemplate does not exists as a k8s kind. it is just an open shift template.

Reply:
@rmohr 
so if its an issue with the overall phrasing  (OfflineVirtualMachine and VirtualMachineTemplate)
changing OfflineVirtualMachine to "virtual machine template" (without mentioning its an ocp template - it is already mentioned at the beginning)  will be the right change ? 

also, are you also suggesting removing OfflineVirtualMachine from "kind:" in code examples ?

Reply:
OfflineVirtualMachine is fine since it exists (in the text and in "kind"). Using "virtual machine template" instead of VirtualMachineTemplate solves my issue.